data-model/src/main/java/io/micronaut/data/model/query/builder/sql/SqlQueryBuilder.java

```diff
@@ -300,9 +302,17 @@ String buildJoinTableInsert(@NonNull PersistentEntity entity, @NonNull Associati
      * @return True if it is.
      */
     public static boolean isForeignKeyWithJoinTable(@NonNull Association association) {
-        return association.isForeignKey() &&
-                !association.getAnnotationMetadata()
-                        .stringValue(Relation.class, "mappedBy").isPresent();
+        if (!association.isForeignKey()) {
+            return false;
+        }
+        if (association.getAnnotationMetadata().stringValue(Relation.class, "mappedBy").isPresent()) {
+            return false;
+        }
+        AnnotationValue<JoinColumns> joinColumnsAnnotationValue  = association.getAnnotationMetadata().getAnnotation(JoinColumns.class);
```

Comment:
This means association is foreign key, has no mappedBy and no join column and join table is needed for foreign key.